### ISSUE TITLE
crash when opening encrypted files that are resized on open

### ISSUE THREAD

**Issue by github2016a** [realm/realm-cocoa#3075]
i have updated xcode to 7.2,and realmswift 0.97.0
but when i do encryption on a existed realm with data ( may generated before 0.97.0), like 

        try! realmOld.writeCopyToPath(newRealm,encryptionKey: getKey())

 when i open the new realm using getKey() , i will always get 
"Error Domain=io.realm Code=2 "Unable to open a realm at path ....."
seems the key is not right, but i am sure that i am using the same key. 

i donot know why. the encryption seems ok before i update to realmswift 0.97.0.

**Comment by jasper-chan** [comment]
I seem to be having a similar problem.

>     RLMRealmConfiguration *configuration = [RLMRealmConfiguration defaultConfiguration];
>    configuration.encryptionKey = [self convertUUIDTo64ByteKey:@"83E28D0C-589F-4248-BB38-14C165D8D5AA"];

>    DDLogInfo(@"%@", configuration.encryptionKey);
>    [RLMRealmConfiguration setDefaultConfiguration:configuration];
>    RLMRealm *defaultRealm = [RLMRealm defaultRealm];

I am able to open the realm file the first time I create the file. If I stop the program, and re-run the exact same code, the app crashes with the exception "Unable to open a realm at path ". If I comment out the encryption key process and re-run the test, I am able to successfully open realm. I am also on XCode 7.2 and realm objective-c 0.97.0

**Comment by mrackwitz** [comment]
As of version 0.97.0, you can use encryption while debugging. We used to recommend adding the environment variable `REALM_DISABLE_ENCRYPTION` to the scheme, you run for debugging your app. This will forcefully disable encryption as feature. Please make sure that you have removed this flag after updating. Realm files, which were generated with previous versions while this flag was enabled, are still unencrypted. You can write an encrypted copy instead as a workaround or just start all over by deleting the file.

**Comment by github2016a** [comment]
yes，i have create a new realm file ,but when the file has little data, the encryption is ok , if has more data like 3M , the same situation appears again. i am not use REALM_DISABLE_ENCRYPTION, and when i archive to app store for test , the problem remain the same.i can not open the new encrypted realm with the key.

**Comment by github2016a** [comment]
i have did some more test, I am quite sure this is a problem, may be a bug...
the test is done in swift-2.1.1/RealmExamples/Encryption.
I have a object called 
```
class MediaSentence: Object {
    dynamic var id :Int = 0
    dynamic var movieId : String = ""
    dynamic var movieType : String = ""
    dynamic var sentenceIdx : Int = 0
    dynamic var subtitle : String = ""
    dynamic var subtitleChi : String = ""
    dynamic var timeStartMS : Int = 0
    dynamic var timeEndMS : Int = 0
    override class func primaryKey() -> String {
        return "id"
    }
    override class func indexedProperties() -> [String] {
        return ["movieId","sentenceIdx","timeStartMS"]
    }
}
```
and a json file has data, I first use the default.realm, import some data, then encrypt to a new realm file , at last open the encrypted file.
```
        let realm = try! Realm()
        importDataFromJsonFile()

        let newRealmPath = getRealPath(["cache.temp.realm"])
        try! realm.writeCopyToPath(newRealmPath, encryptionKey: getKey())
        
        let config2 = Realm.Configuration(
            path: newRealmPath,
            encryptionKey: getKey()
        )
        print("before open")
        try! Realm(configuration: config2) // this may throw exception "Unable to open a realm at path..."
        print("after open")
```
the result is : 
when the json table resords is less than 600( the realm file is less than 150kb), everything going fine.
when the json table resords is more than 1000( the realm file is more than 250kb), the problem appears. when i open the encrypted file with key, it throws : 
```
         fatal error: 'try!' expression unexpectedly raised an error: Error Domain=io.realm Code=2 "Unable to open a realm at path '
```
so i am sure the exception appears or not , depends on the size of realm file.
before i update to realmswift 0.97.0, everything is ok. and this problem has annoyed me several days, and also took some time to deal with it. I hope you guys can help me to find a solution . with this problem , i cannot archive new app version to app store at all.

**Comment by github2016a** [comment]
:pray:

**Comment by mrackwitz** [comment]
Could you share the JSON file with us? (privately to help@realm.io) Then we could try to reproduce the issue.

**Comment by github2016a** [comment]
yes,of course, i will send the test project.

**Comment by mrackwitz** [comment]
Thanks, just confirming that we received the files.

**Comment by github2016a** [comment]
@mrackwitz i have send the data to help@realm.io, you can check it when you have time, if there is any problem please let me know, thanks very much!

**Comment by github2016a** [comment]
@mrackwitz i have send the data to help@realm.io, you can check it when you have time, if there is any problem please let me know, thanks very much!

**Comment by github2016a** [comment]
@mrackwitz i have send the data to help@realm.io, you can check it when you have time, if there is any problem please let me know, thanks very much!

**Comment by github2016a** [comment]
@mrackwitz i have send the data to help@realm.io, you can check it when you have time, if there is any problem please let me know, thanks very much!

**Comment by github2016a** [comment]
@mrackwitz i have send the data to help@realm.io, you can check it when you have time, if there is any problem please let me know, thanks very much!

**Comment by github2016a** [comment]
@mrackwitz i have send the data to help@realm.io, you can check it when you have time, if there is any problem please let me know, thanks very much!

**Comment by github2016a** [comment]
@mrackwitz i have send the data to help@realm.io, you can check it when you have time, if there is any problem please let me know, thanks very much!

**Comment by tgoyne** [comment]
Thanks for the repro case. There's a bug in the underlying core library which I've now made a PR to fix (for certain sizes of encrypted files the opening logic looks in the wrong place for the file footer). I don't think there's a reasonable workaround other than using a version older than 0.96.3 until we can ship a fix.

**Comment by github2016a** [comment]
Ok,i see.i will use older version for now, and wait for the fix, thanks.

**Comment by github2016a** [comment]
for swiftrealm 0.96.3，there is error too
Error Domain=io.realm Code=2 "Bad Realm file header (#1)
for swiftrealm 0.96.2，everything ok！

**Comment by jpsim** [comment]
A fix for this was merged in #3015.
